src/window.py

```diff
@@ -0,0 +1,154 @@
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+from gi.repository import Gtk, Gio, GLib
+from .updutillogger import UpdutilLogger
+import os
+
+@Gtk.Template(resource_path='/org/worldpossible/updutil/window.ui')
+class WorldpossibleUpdutilWindow(Gtk.ApplicationWindow):
+    __gtype_name__ = 'WorldpossibleUpdutilWindow'
+
+    _log = UpdutilLogger()
+
+    chooser_button = Gtk.Template.Child()
+    success_result_label = Gtk.Template.Child()
+    failure_result_label = Gtk.Template.Child()
+    output_label = Gtk.Template.Child()
+    output_window = Gtk.Template.Child()
+    output_buffer = Gtk.Template.Child()
+    path_entry = Gtk.Template.Child()
+    save_button = Gtk.Template.Child()
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.chooser_button.connect('clicked', self.on_chooser_clicked)
+        self.save_button.connect('clicked', self.on_save_clicked)
+        self.show_warning_dialog()
+
+    def show_warning_dialog(self):
+        dialog = Gtk.MessageDialog(self,
+                                   Gtk.DialogFlags.MODAL,
+                                   Gtk.MessageType.INFO,
+                                   Gtk.ButtonsType.OK,
+                                   'This utility allows you to execute scripts with administrator privileges!')
+        dialog.format_secondary_text('Only choose files from trusted sources to avoid harming your computer.')
+        dialog.run()
+        dialog.destroy()
+
+    def reset(self):
+        self.success_result_label.hide()
+        self.success_result_label.set_text('')
+        self.failure_result_label.hide()
+        self.failure_result_label.set_text('')
+        self.output_label.hide()
+        self.output_window.hide()
+        self.output_buffer.set_text('')
+        self.path_entry.set_text('')
+        self.save_button.hide()
+
+    def on_chooser_clicked(self, button):
+        self.reset()
+
+        chooser = Gtk.FileChooserNative.new('Choose Script', self,
+                                            Gtk.FileChooserAction.OPEN,
+                                            'Run', None)
+        res = chooser.run()
+        if res != Gtk.ResponseType.ACCEPT:
+            return
+
+        path = chooser.get_filename()
+        self.path_entry.set_text(path)
+
+        if not os.access(path, os.X_OK):
+            self.failure_result_label.set_text('Error executing script: executable permission not set')
+            self.failure_result_label.show()
+            return
```

Comment:
Yeah, invoking bash seems like a pretty good solution. It means this tool is a bit less broadly useful since it won't work as well with, say, python scripts, but one could still use a python script if it's made executable, and shell scripts on a USB drive are probably the common use case.